### PR TITLE
[Merged by Bors] - feat: port RingTheory.Polynomial.Chebyshev

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by int-y1** [comment]
todo:

- [ ] fix `T_mul`. goal is `⊢ 2 = comp 2 (T R n)`

**Comment by RemyDegenne** [comment]
Thanks!
bors r+

File: Mathlib/Data/Polynomial/Eval.lean [hunk 1]
```diff
@@ -746,6 +751,7 @@ def mapRingHom (f : R →+* S) : R[X] →+* S[X] where
   map_one' := Polynomial.map_one f
 #align polynomial.map_ring_hom Polynomial.mapRingHom
 
+
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Polynomial/Eval.lean:754]:
```suggestion
```

**Comment by Ruben-VandeVelde** [comment]
Looks like this was the cause of the failure:
```

./././Mathlib/RingTheory/Polynomial/Chebyshev.lean:281:78: error: unsolved goals
case h.e'_2.h.e'_6.h.e'_3.h.e
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ HMul.hMul = HAdd.hAdd

case h.e'_2.h.e'_6.h.e'_3.h.e'_1
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ 1 + m = n

case h.e'_2.h.e'_6.h.e'_3.h.e'_2
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ n = m * n

case h.e'_3.h.e'_5.h.e'_3.h.e
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ HMul.hMul = HAdd.hAdd

case h.e'_3.h.e'_5.h.e'_3.h.e'_1.h.e
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ HAdd.hAdd = HMul.hMul

case h.e'_3.h.e'_5.h.e'_3.h.e'_1.h.e'_1
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ m = 2

case h.e'_3.h.e'_5.h.e'_3.h.e'_1.h.e'_2
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ 2 = n

case h.e'_3.h.e'_5.h.e'_3.h.e'_2
R : Type u_1
S : Type ?u.135667
inst✝¹ : CommRing R
inst✝ : CommRing S
m n : ℕ
⊢ n = m * n
Try this: ring_nf
Try this: ring_nf
Try this: ring_nf
Try this: ring_nf
Try this: ring_nf
Try this: ring_nf
Try this: ring_nf
Try this: ring_nf
```

**Comment by mo271** [comment]
Tried to change `ring` to `ring_nf` as a fix... @Ruben-VandeVelde

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**